akka-actor/src/main/scala/akka/io/TcpConnection.scala

```diff
@@ -360,16 +360,25 @@ private[io] abstract class TcpConnection(val tcp: TcpExt, val channel: SocketCha
     create(write)
   }
 
-  def PendingBufferWrite(commander: ActorRef, data: ByteString, ack: Event, tail: WriteCommand): PendingBufferWrite = {
-    val buffer = bufferPool.acquire()
-    try {
-      val copied = data.copyToBuffer(buffer)
-      buffer.flip()
-      new PendingBufferWrite(commander, data.drop(copied), ack, buffer, tail)
-    } catch {
-      case NonFatal(e) ⇒
-        bufferPool.release(buffer)
-        throw e
+  //TODO: move canWrapAsByteBuffer into PendingBufferWrite#doWrite
+  //TODO: check chunk size
+  def PendingBufferWrite(commander: ActorRef, data: ByteString, ack: Any, tail: WriteCommand): PendingBufferWrite = {
```

Comment:
this was mostly because the class takes an ack: Any, but the factory method took an ack: Event (and couldnt then call the factory method from the class, which has the 'bytebuffer picking' logic)

I can probably align to either ack: Any or ack: Event. I've noticed the api for like akka.io.udp asks for an ack of type 'Event', but I've picked through the logic and it didnt really seem required anywhere. its not sealed or anything, but it might also be overreach for this pr.